### DISCUSSION TITLE
数学公式中正体字体问题

### DISCUSSION THREAD

**Discussion by jzmzz** [note286/xduts#55] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/63654020/222025633-7b0411c1-1e26-4872-b5a3-b054d1d1e2c8.png)
如图，第一个sin处于文本模式下，第二个sin为设置了`math-font = bonum`之后的数学模式显示效果，第三个为手动添加命令`\setmathrm{TeX Gyre Bonum Math}`之后的显示效果，因此数学模式下的正体`\mathrm`似乎不受`math-font = bonum`的影响，它的字体应该被是`\setmainfont`控制了。
不知道是不是bug，选择bonum字体只为突出问题，使用Computer Modern区别没那么明显，但可以感觉到风格明显不统一。

**Comment by note286** [comment, marked as answer]
手册中已经提及，除了Computer Modern数学字体外，均为`unicode-math`宏包处理，因此，如果你想使用正体，请使用`\symup`而不是`\mathrm`。此外，`\symrm`和`\symup`是互为别名，这一点你可以在`unicode-math`手册中看到。

在`unicode-math`中，`\math`系列的命令是调用文本字体，而`\sym`系列的命令是调用数学字体，同样地，这一点你可以在`unicode-math`手册中看到。因此你可以看到下面的示例中，`\symup{x}`和` \mathrm{x}`的显示效果不同，是因为不是同一个字体。

一般来说，公式符号为意大利体，这是约定俗成的，你在LaTeX直接写公式默认就是意大利体。因此，我建议你使用`$\sin x$`而不是`$\sin \symup{x}$`。

最后，由于学位论文规定英文字体为Times New Roman，因此我更建议你选择一款Times风格的数学字体，例如XITS Math。

```latex
\documentclass{xdupgthesis}
\xdusetup{style/math-font=bonum}
\begin{document}
$\sin x$
$\sin \symup{x}$
$\sin \mathrm{x}$
\end{document}
```

![image](https://user-images.githubusercontent.com/16207292/222036531-a46fcf42-a988-41dd-bd4f-fd2791a01352.png)

**Reply by jzmzz** [reply]
问题在于sin的字体，而不是x的字体，在论文中使用`$\sin x$`sin的字体就和x的字体不一样了，按理说sin 和x的字体风格应该是统一的，但是你看图中 sin 和x的风格明显不是一个

**Reply by note286** [reply]
下次记得描述清楚，大不了多说一些冗余信息。

```latex
\documentclass{xdupgthesis}
\xdusetup{style/math-font=bonum}
\AtBeginDocument{\setoperatorfont\symup}
\begin{document}
sin x
$\sin x$
$\sin \symup{x}$
$\sin \mathrm{x}$
\end{document}
```

![image](https://user-images.githubusercontent.com/16207292/222041079-376bff0d-50f1-4237-bbf0-6192b2b282b7.png)

关于算子的字体问题，传统数学那一套使用的是正文字体，`unicode-math`默认使用的也是正文字体，你可以切换为数学字体，甚至切换样式，这一点`unicode-math`手册已经提及了。

![image](https://user-images.githubusercontent.com/16207292/222041245-caa4a89c-51c9-4612-96bd-f8cc6cf19228.png)

**Reply by note286** [reply]
我仔细琢磨了一下你这一句

> 按理说sin和x的字体风格应该是统一的

你或许需要温习一下`算符`的概念，还有`DeclareMathOperator`，以及`\sin`到底是什么？

![image](https://user-images.githubusercontent.com/16207292/222044009-11c1b622-f466-400c-90a8-e953e9d01b59.png)

**Reply by jzmzz** [reply]
其实问题起因是，我发现xduts的数学模式下`\sin x`中的sin比默认latex的`\sin x`粗了一点，所以就有了这个问题，现在了解了。

**Reply by note286** [reply]
不知道你所谓的默认是指什么？如果是`article`之类的文档类，不加载额外的数学字体宏包的话，就是Computer Modern字体，XDUTS默认的数学字体也是Computer Modern，是一致的，没有任何区别。

**Reply by sikouhjw** [reply]
> 不知道你所谓的默认是指什么？如果是`article`之类的文档类，不加载额外的数学字体宏包的话，就是Computer Modern字体，XDUTS默认的数学字体也是Computer Modern，是一致的，没有任何区别。

他指的是算符的字体不同，模板用的应该是 `XITS`，所以有区别。

**Reply by jzmzz** [reply]
ctexbook，未加载任何包，数学模式下，xduts,也是默认设置，sin的粗细确实有不同。

**Reply by jzmzz** [reply]
我现在知道原因了，但我觉得这个设置弄成默认的比较好吧，不然sin使用Times字体，x使用Computer Modern字体，会不会不协调？

**Reply by sikouhjw** [reply]
> 我现在知道原因了，但我觉得这个设置弄成默认的比较好吧，不然sin使用Times字体，x使用Computer Modern字体，会不会不协调？

1. 正文字体显然是学校的规定
    https://github.com/note286/xduts/blob/c131e907b87902feaf5186fee99c6631f7ae90ac/xduts.dtx#L556-L562
2. 模板的原则是『不教用户做事』，你可以自行设置
3. 有没有可能，数学用 Computer Modern 本身就不协调？为了协调你是不是应该用 `XITS Math` 比较好？

**Reply by note286** [reply]
> ctexbook，未加载任何包，数学模式下，xduts,也是默认设置，sin的粗细确实有不同。
>
> 我现在知道原因了，但我觉得这个设置弄成默认的比较好吧，不然sin使用Times字体，x使用Computer Modern字体，会不会不协调？

`ctexbook`文档类下，正文是Computer Modern字体。

XDUTS正文字体默认是Times风格的，是通过`fontspec`设置的，之前没有加载`no-math`选项，目前已经默认设置`no-math`选项。

```latex
\documentclass{xdupgthesis}
\begin{document}
sin x
$\sin x$
\end{document}
```

![image](https://user-images.githubusercontent.com/16207292/222065268-118cfa21-e6f2-4845-a5af-fd404a24f716.png)


此外，目前使用`unicode-math`时，算符字体默认也为数学字体。

```latex
\documentclass{xdupgthesis}
\xdusetup{style/math-font=lm}
\begin{document}
sin x
$\sin x$
\end{document}
```

![image](https://user-images.githubusercontent.com/16207292/222065184-bcd75eae-96e7-4997-bc19-64915673b08c.png)

```latex
\documentclass{xdupgthesis}
\xdusetup{style/math-font=bonum}
\begin{document}
sin x
$\sin x$
\end{document}
```

![image](https://user-images.githubusercontent.com/16207292/222065420-a7f79f87-74be-47bf-860f-547e613b8c0c.png)


---

关于数学字体，多说几句。

一般来说，正文字体和数学字体风格一致是比较合适的。但是由于学校强制规定正文字体为Times New Roman，因此，选择一个Times风格的数学字体才是一个好的选择。

数学字体方面，大部分用户接触的都是传统数学的math alphabets，字体的话甚至只有Computer Modern字体。很多老师或者学生阅读Computer Modern字体已经习惯了，其他的字体（哪怕是Times风格的数学字体）都不喜欢。这就会导致正文中英文和数学的风格很不一致，而且最明显的差异还是粗细。

目前较为好的选择就是使用`unicode-math`，可以保证数学字体风格的一致性。如果使用math alphabets，缺符号就要找包，每个包的字体风格还不一致。有些时候，即使用户只想用这个包中的若干个符号，但是很多数学字体包还会覆盖其他符号的字体，目前我只发现了`stmaryrd`提供了`only`参数，用来限定符号生效范围。使用`unicode-math`并选择一个合适的字体就可以规避这个问题。不过有多少用户知道`unicode-math`？如果默认使用了`unicode-math`，那么很多用户连符号加粗这么常见的操作都不会，又有多少用户肯阅读`unicode-math`手册呢？

不过，能用总比不会用强。因此，XDUTS默认值的设置优先考虑的不是美观，是易用性。具体到数学字体方面，就是没有做任何修改，让用户无须改变之前的使用习惯，哪怕和正文英文字体很不搭配。

**Reply by sikouhjw** [reply]
事实上他们一般用的加粗都是 `\mathbf` 之类的命令，`unicode-math` 是兼容这些命令的，还可以在模板中重定义这些命令，再在正文中使用来『教』他们用。当然这也许不符合模板的理念吧

**Reply by note286** [reply]
> 他指的是算符的字体不同，模板用的应该是 `XITS`，所以有区别。

@sikouhjw 我`fontspec`没加`no-math`参数，导致算符的字体为正文字体，默认数学字体还是Computer Modern，所以看起来就不是很协调，已经把`no-math`加上了。

**Reply by note286** [reply]
> 事实上他们一般用的加粗都是 `\mathbf` 之类的命令，`unicode-math` 是兼容这些命令的，还可以在模板中重定义这些命令，再在正文中使用来『教』他们用。当然这也许不符合模板的理念吧

@sikouhjw 是兼容，不过这样走的是正文字体，其实不是我们想要的。不过无所谓了，在意的人自然看看文档就知道了，不在意的人又何必主动教他们呢？虽然没有统计数据，但是我感觉还是Computer Modern用的人多。

**Reply by note286** [reply]
> 当然这也许不符合模板的理念吧

@sikouhjw 你说的很对，手册就应该描述独有的用法或者与一般用法不一致的地方，顶多指个路。我开了一个讨论区 #40，已经是我最大的妥协了。
